{
  "post_stream": {
    "posts": [
      {
        "id": 9967,
        "name": "Peter Zeller",
        "username": "peter",
        "avatar_template": "/user_avatar/forums.foundationdb.org/peter/{size}/1064_2.png",
        "created_at": "2021-09-14T09:14:48.041Z",
        "cooked": "<p>When I try to restore my database from a backup using the Kubernetes operator I get the following error in the operator log:</p>\n<pre><code class=\"lang-auto\">{\"level\":\"info\",\"ts\":1631609197.321,\"logger\":\"fdbclient\",\"msg\":\"Running command\",\"namespace\":\"verifier\",\"cluster\":\"sample-cluster\",\"path\":\"/usr/bin/fdb/6.3/fdbrestore\",\"args\":[\"/usr/bin/fdb/6.3/fdbrestore\",\"start\",\"-r\",\"blobstore://s3-proxy-user@fdb-backup-s3-proxy:80/default?bucket=foundationdbbackup&amp;secure_connection=0\",\"--dest_cluster_file\",\"/tmp/405498555\",\"--log\",\"--logdir\",\"/var/log/fdb\"]}\n{\"level\":\"error\",\"ts\":1631609207.3601973,\"logger\":\"fdbclient\",\"msg\":\"Error from FDB command\",\"namespace\":\"verifier\",\"cluster\":\"sample-cluster\",\"code\":-1,\"stdout\":\"No restore target version given, will use maximum restorable version from backup description.\\nUsing target restore version 1207990314200\\n\",\"stderr\":\"\",\"error\":\"signal: killed\",\"stacktrace\":\"github.com/go-logr/zapr.(*zapLogger).Error\\n\\t/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\\nsigs.k8s.io/controller-runtime/pkg/log.(*DelegatingLogger).Error\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/log/deleg.go:144\\ngithub.com/FoundationDB/fdb-kubernetes-operator/fdbclient.(*cliAdminClient).runCommand\\n\\t/workspace/fdbclient/admin_client.go:187\\ngithub.com/FoundationDB/fdb-kubernetes-operator/fdbclient.(*cliAdminClient).StartRestore\\n\\t/workspace/fdbclient/admin_client.go:555\\ngithub.com/FoundationDB/fdb-kubernetes-operator/controllers.StartRestore.Reconcile\\n\\t/workspace/controllers/start_restore.go:48\\ngithub.com/FoundationDB/fdb-kubernetes-operator/controllers.(*FoundationDBRestoreReconciler).Reconcile\\n\\t/workspace/controllers/restore_controller.go:77\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216\\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185\\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:155\\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:156\\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:133\\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185\\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:99\"}\n{\"level\":\"error\",\"ts\":1631609207.3605845,\"logger\":\"controller\",\"msg\":\"Error in reconciliation\",\"namespace\":\"verifier\",\"restore\":\"sample-cluster\",\"subReconciler\":\"controllers.StartRestore\",\"requeueAfter\":0,\"error\":\"signal: killed\",\"stacktrace\":\"github.com/go-logr/zapr.(*zapLogger).Error\\n\\t/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\\ngithub.com/FoundationDB/fdb-kubernetes-operator/controllers.processRequeue\\n\\t/workspace/controllers/controllers.go:114\\ngithub.com/FoundationDB/fdb-kubernetes-operator/controllers.(*FoundationDBRestoreReconciler).Reconcile\\n\\t/workspace/controllers/restore_controller.go:82\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216\\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185\\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:155\\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:156\\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:133\\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185\\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:99\"}\n{\"level\":\"error\",\"ts\":1631609207.3613954,\"logger\":\"controller-runtime.manager.controller.foundationdbrestore\",\"msg\":\"Reconciler error\",\"reconciler group\":\"apps.foundationdb.org\",\"reconciler kind\":\"FoundationDBRestore\",\"name\":\"sample-cluster\",\"namespace\":\"verifier\",\"error\":\"signal: killed\",\"stacktrace\":\"github.com/go-logr/zapr.(*zapLogger).Error\\n\\t/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:302\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253\\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\\n\\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216\\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185\\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:155\\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:156\\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:133\\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185\\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\\n\\t/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:99\"}\n</code></pre>\n<p>When I run the fdbrestore command manually (from the backup agent pod) the restore works without problems.</p>\n<p>The URL <code>s3-proxy-user@fdb-backup-s3-proxy</code> I am using is a proxy to provide an S3 API for Azure blob storage.</p>\n<p>Any ideas what might be the problem here or how I can debug this?</p>",
        "post_number": 1,
        "post_type": 1,
        "posts_count": 2,
        "updated_at": "2021-09-14T09:14:48.041Z",
        "reply_count": 0,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 549,
        "reads": 21,
        "readers_count": 20,
        "score": 2744.2,
        "yours": false,
        "topic_id": 2886,
        "topic_slug": "error-signal-killed-when-recovering-from-backup",
        "display_username": "Peter Zeller",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": null,
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 927,
        "hidden": false,
        "trust_level": 1,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/error-signal-killed-when-recovering-from-backup/2886/1",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null,
        "can_vote": false
      },
      {
        "id": 9970,
        "name": "Johannes Scheuermann",
        "username": "johscheuer",
        "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/j/f475e1/{size}.png",
        "created_at": "2021-09-15T14:26:53.140Z",
        "cooked": "<p>I would spin up a foundationdb Pod with the same version and mount the cluster file. Then exec into this container and run the same command and hopefully in the trace logs there should be more information. If you already collect the trace logs of the operator Pod you can also look at them directly. One thing that comes to my mind (since this is a kill) that the default timeout is hit and the go program shutdown the process. If you can run the command in the separate Pod without any issues I would assume that this is the problem. Given the ~10 seconds between the first log line and the second line with the kill (and you\u2019re right the default timeout is 10 seconds), that\u2019s probably the case. Feel free to create an issue in GitHub and file a PR that resolves that issue. The backup and restore feature is still in an alpha stage.</p>",
        "post_number": 2,
        "post_type": 1,
        "posts_count": 2,
        "updated_at": "2021-09-15T14:26:53.140Z",
        "reply_count": 0,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 1,
        "reads": 17,
        "readers_count": 16,
        "score": 23.4,
        "yours": false,
        "topic_id": 2886,
        "topic_slug": "error-signal-killed-when-recovering-from-backup",
        "display_username": "Johannes Scheuermann",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "bookmarked": false,
        "actions_summary": [
          {
            "id": 2,
            "count": 1
          }
        ],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 849,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/error-signal-killed-when-recovering-from-backup/2886/2",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      }
    ],
    "stream": [
      9967,
      9970
    ]
  },
  "timeline_lookup": [
    [
      1,
      1499
    ],
    [
      2,
      1497
    ]
  ],
  "suggested_topics": [],
  "tags": [],
  "tags_descriptions": {},
  "fancy_title": "Error &ldquo;signal: killed&rdquo; when recovering from backup",
  "id": 2886,
  "title": "Error \"signal: killed\" when recovering from backup",
  "posts_count": 2,
  "created_at": "2021-09-14T09:14:47.962Z",
  "views": 1141,
  "reply_count": 0,
  "like_count": 1,
  "last_posted_at": "2021-09-15T14:26:53.140Z",
  "visible": true,
  "closed": false,
  "archived": false,
  "has_summary": false,
  "archetype": "regular",
  "slug": "error-signal-killed-when-recovering-from-backup",
  "category_id": 16,
  "word_count": 1286,
  "deleted_at": null,
  "user_id": 927,
  "featured_link": null,
  "pinned_globally": false,
  "pinned_at": null,
  "pinned_until": null,
  "image_url": null,
  "slow_mode_seconds": 0,
  "draft": null,
  "draft_key": "topic_2886",
  "draft_sequence": null,
  "unpinned": null,
  "pinned": false,
  "current_post_number": 1,
  "highest_post_number": 2,
  "deleted_by": null,
  "actions_summary": [
    {
      "id": 4,
      "count": 0,
      "hidden": false,
      "can_act": false
    },
    {
      "id": 8,
      "count": 0,
      "hidden": false,
      "can_act": false
    },
    {
      "id": 10,
      "count": 0,
      "hidden": false,
      "can_act": false
    },
    {
      "id": 7,
      "count": 0,
      "hidden": false,
      "can_act": false
    }
  ],
  "chunk_size": 20,
  "bookmarked": false,
  "topic_timer": null,
  "message_bus_last_id": 0,
  "participant_count": 2,
  "show_read_indicator": false,
  "thumbnails": null,
  "slow_mode_enabled_until": null,
  "tags_disable_ads": false,
  "related_topics": [
    {
      "fancy_title": "FoundationDBBackup backup command error",
      "id": 2341,
      "title": "FoundationDBBackup backup command error",
      "slug": "foundationdbbackup-backup-command-error",
      "posts_count": 5,
      "reply_count": 3,
      "highest_post_number": 5,
      "image_url": null,
      "created_at": "2020-09-17T20:19:07.206Z",
      "last_posted_at": "2020-09-17T20:58:51.172Z",
      "bumped": true,
      "bumped_at": "2020-09-19T00:01:34.791Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 0,
      "views": 657,
      "category_id": 16,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": null,
          "description": "Original Poster",
          "user": {
            "id": 819,
            "username": "mbuleandra",
            "name": "Marius",
            "avatar_template": "/user_avatar/forums.foundationdb.org/mbuleandra/{size}/977_2.png",
            "trust_level": 1
          }
        },
        {
          "extras": "latest",
          "description": "Most Recent Poster",
          "user": {
            "id": 3,
            "username": "john_brownlee",
            "name": "John Brownlee",
            "avatar_template": "/user_avatar/forums.foundationdb.org/john_brownlee/{size}/22_2.png",
            "admin": true,
            "trust_level": 4
          }
        }
      ]
    },
    {
      "fancy_title": "Failure / Recovery scenario",
      "id": 2384,
      "title": "Failure / Recovery scenario",
      "slug": "failure-recovery-scenario",
      "posts_count": 2,
      "reply_count": 0,
      "highest_post_number": 2,
      "image_url": null,
      "created_at": "2020-10-10T01:06:26.706Z",
      "last_posted_at": "2020-10-12T13:36:11.016Z",
      "bumped": true,
      "bumped_at": "2020-10-12T13:36:11.016Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 0,
      "views": 695,
      "category_id": 16,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": null,
          "description": "Original Poster",
          "user": {
            "id": 819,
            "username": "mbuleandra",
            "name": "Marius",
            "avatar_template": "/user_avatar/forums.foundationdb.org/mbuleandra/{size}/977_2.png",
            "trust_level": 1
          }
        },
        {
          "extras": "latest",
          "description": "Most Recent Poster",
          "user": {
            "id": 3,
            "username": "john_brownlee",
            "name": "John Brownlee",
            "avatar_template": "/user_avatar/forums.foundationdb.org/john_brownlee/{size}/22_2.png",
            "admin": true,
            "trust_level": 4
          }
        }
      ]
    },
    {
      "fancy_title": "Restoring the backup that was taken through fdbbackup results in database lock",
      "id": 1849,
      "title": "Restoring the backup that was taken through fdbbackup results in database lock",
      "slug": "restoring-the-backup-that-was-taken-through-fdbbackup-results-in-database-lock",
      "posts_count": 3,
      "reply_count": 1,
      "highest_post_number": 3,
      "image_url": null,
      "created_at": "2019-12-17T19:41:16.851Z",
      "last_posted_at": "2019-12-18T18:34:06.575Z",
      "bumped": true,
      "bumped_at": "2019-12-18T18:34:06.575Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 0,
      "views": 853,
      "category_id": 7,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": "latest",
          "description": "Original Poster, Most Recent Poster",
          "user": {
            "id": 700,
            "username": "viveksinghggits",
            "name": "Vivek Singh",
            "avatar_template": "/user_avatar/forums.foundationdb.org/viveksinghggits/{size}/737_2.png",
            "trust_level": 1
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 7,
            "username": "SteavedHams",
            "name": "Steve Atherton",
            "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
            "trust_level": 4
          }
        }
      ]
    },
    {
      "fancy_title": "Getting timeout error while reconciling foundationDB cluster in kubernetes",
      "id": 2150,
      "title": "Getting timeout error while reconciling foundationDB cluster in kubernetes",
      "slug": "getting-timeout-error-while-reconciling-foundationdb-cluster-in-kubernetes",
      "posts_count": 2,
      "reply_count": 0,
      "highest_post_number": 2,
      "image_url": null,
      "created_at": "2020-06-01T10:44:55.604Z",
      "last_posted_at": "2020-06-01T16:46:30.092Z",
      "bumped": true,
      "bumped_at": "2020-06-01T16:46:30.092Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 0,
      "views": 789,
      "category_id": 16,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": null,
          "description": "Original Poster",
          "user": {
            "id": 772,
            "username": "antonprabin",
            "name": "Prabin Poulose",
            "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/a/7cd45c/{size}.png",
            "trust_level": 1
          }
        },
        {
          "extras": "latest",
          "description": "Most Recent Poster",
          "user": {
            "id": 3,
            "username": "john_brownlee",
            "name": "John Brownlee",
            "avatar_template": "/user_avatar/forums.foundationdb.org/john_brownlee/{size}/22_2.png",
            "admin": true,
            "trust_level": 4
          }
        }
      ]
    },
    {
      "fancy_title": "Regarding issue faced while running fdbbackup",
      "id": 1458,
      "title": "Regarding issue faced while running fdbbackup",
      "slug": "regarding-issue-faced-while-running-fdbbackup",
      "posts_count": 17,
      "reply_count": 14,
      "highest_post_number": 17,
      "image_url": null,
      "created_at": "2019-06-17T12:33:04.574Z",
      "last_posted_at": "2019-12-17T19:51:11.604Z",
      "bumped": true,
      "bumped_at": "2019-12-17T19:51:11.604Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 1,
      "views": 2007,
      "category_id": 7,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": null,
          "description": "Original Poster",
          "user": {
            "id": 587,
            "username": "pragawaran",
            "name": "",
            "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/p/f07891/{size}.png",
            "trust_level": 1
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 166,
            "username": "gaurav",
            "name": "gaurav",
            "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/g/b487fb/{size}.png",
            "trust_level": 3
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 12,
            "username": "ajbeamon",
            "name": "A.J. Beamon",
            "avatar_template": "/user_avatar/forums.foundationdb.org/ajbeamon/{size}/13_2.png",
            "admin": true,
            "trust_level": 4
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 700,
            "username": "viveksinghggits",
            "name": "Vivek Singh",
            "avatar_template": "/user_avatar/forums.foundationdb.org/viveksinghggits/{size}/737_2.png",
            "trust_level": 1
          }
        },
        {
          "extras": "latest",
          "description": "Most Recent Poster",
          "user": {
            "id": 3,
            "username": "john_brownlee",
            "name": "John Brownlee",
            "avatar_template": "/user_avatar/forums.foundationdb.org/john_brownlee/{size}/22_2.png",
            "admin": true,
            "trust_level": 4
          }
        }
      ]
    }
  ],
  "summarizable": false,
  "can_vote": false,
  "vote_count": 0,
  "user_voted": false,
  "discourse_zendesk_plugin_zendesk_id": null,
  "discourse_zendesk_plugin_zendesk_url": "https://your-url.zendesk.com/agent/tickets/",
  "details": {
    "can_edit": false,
    "notification_level": 1,
    "participants": [
      {
        "id": 849,
        "username": "johscheuer",
        "name": "Johannes Scheuermann",
        "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/j/f475e1/{size}.png",
        "post_count": 1,
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_color": null,
        "flair_bg_color": null,
        "flair_group_id": null,
        "trust_level": 2
      },
      {
        "id": 927,
        "username": "peter",
        "name": "Peter Zeller",
        "avatar_template": "/user_avatar/forums.foundationdb.org/peter/{size}/1064_2.png",
        "post_count": 1,
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_color": null,
        "flair_bg_color": null,
        "flair_group_id": null,
        "trust_level": 1
      }
    ],
    "created_by": {
      "id": 927,
      "username": "peter",
      "name": "Peter Zeller",
      "avatar_template": "/user_avatar/forums.foundationdb.org/peter/{size}/1064_2.png"
    },
    "last_poster": {
      "id": 849,
      "username": "johscheuer",
      "name": "Johannes Scheuermann",
      "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/j/f475e1/{size}.png"
    }
  },
  "bookmarks": []
}